ssl/statem/statem_lib.c

```diff
@@ -1004,6 +1005,73 @@ int ssl_choose_server_version(SSL *s, CLIENTHELLO_MSG *hello)
         break;
     }
 
+    suppversions = tls_get_extension_by_type(hello->pre_proc_exts,
+                                             hello->num_extensions,
+                                             TLSEXT_TYPE_supported_versions);
+
```

Comment:
Yes, that's what it says must be sent for a TLS1.3 client, but what should happen if a TLS1.1. client sends it? Presumably it should set it to TLS1.1 because many servers will not understand the extension. Therefore I think we have to ignore the legacy client version altogether if supported_versions is present.